File: AtCoder/ABC177/E.cpp
#include <iostream>
#include <vector>
#include <string>
#include <numeric>

using namespace std;

string solve(const vector<int>& a){
    int v = a[0];
    for(auto& t : a) v = gcd(v, t);
    if(v > 1) return "not coprime";
    vector<int> m(1000000, 0);
    for(auto& t : a){
        int v = t;
        for(int d=2;d*d<=v;d++){
            if(v%d) continue;
            if(m[d]) return "setwise coprime";
            while(v%d == 0) v /= d;
            m[d] = 1;
        }
        if(v > 1){
            if(m[v]) return "setwise coprime";
            m[v] = 1;
        }
    }
    return "pairwise coprime";
}

int main(){
    int N; cin >> N;
    vector<int> a(N);
    for(auto& t : a) cin >> t;
    cout << solve(a) << endl;
}

File: AtCoder/ABC177/E.rs
fn solve(a: Vec<i64>) -> &'static str {
    if a.iter().fold(a[0], |x, &y| num::integer::gcd(x, y)) > 1 {
        return "not coprime"
    }
    let mut m = vec![0; 1000000];
    for &t in &a {
        let mut v = t as usize;
        for div in 2..v {
            if div*div > v { break }
            if v%div != 0 { continue }
            if m[div] != 0 { return "setwise coprime" }
            while v%div == 0 { v /= div }
            m[div] = 1
        }
        if v > 1 {
            if m[v] != 0 { return "setwise coprime" }
            m[v] = 1
        }
    }
    "pairwise coprime"
}

fn main() {
    proconio::input! {
        n: usize,
        a: [i64; n]
    }
    println!("{}", solve(a));
}
